AirCasting/AirCastingApp.swift

```diff
@@ -63,6 +63,15 @@ final class SynchronizationScheduler {
         authorization
             .$isLoggedIn
             .removeDuplicates()
+            .scan((nil, nil)) {
```

Comment:
Could you explain more or less what does it do?

Reply:
of course :)

I noticed this authorization observer will trigger even when authorization is set for the first time during runtime. This happens right when the app launches - for that we already have an observer. To make this clearer, step by step scenario of this false-positive case:
1. User launches the app and they've been already logged in during previous session
2. We notice that and set `authorization.isLoggedin` to `true` as we should.
3. This observer gets triggered. And it should not since the user didn't actually perform a login.

This one should **only** trigger when we switch from logged-out state to logged-in state. To accomplish that I'm using [scan(_:_:)](https://developer.apple.com/documentation/combine/fail/scan(_:_:)) in order to transform a `Bool` publisher into a `(Bool, Bool)` publisher which contains previous value (`$0.0`) and the newly set one (`$0.1`) so we can detect desired scenario.